### ISSUE TITLE
[feature] Pebble integration

### ISSUE THREAD

**Issue by rubdos** [iRail/harbour-berail#44]
Hey Dylan!

This is a feature request. I'll soon be having a Pebble Time Round on my wrist, and it would be *awesome* if I got my trains (and delays!! #14 is related) on my watch in the Pebble timeline.

Thanks to the [rockpool](https://github.com/abranson/rockpool/) application by @abranson, this works decently on SailfishOS.
Rockpool has a daemon that listens to certain DBUS interfaces (@abranson, correct me if I'm wrong please!), one of which is the `insertTimelinePin` call. https://github.com/abranson/rockpool/blob/master/rockworkd/dbusinterface.cpp#L166 lists an example call: `dbus-send --session --dest=org.rockwork --type=method_call --print-reply /org/rockwork/XX_XX_XX_XX_XX_XX org.rockwork.Pebble.insertTimelinePin string:"$(cat pin.json)"`.

It looks like [D-Bus](https://sailfishos.org/develop/docs/nemo-qml-plugin-dbus/) is quite easily interfaces in SailfishOS QML.

I'm mostly opening this issue to ask:

1. @DylanVanAssche, would you accept a patch to make this work?
2. @abranson, would you be okay with me "abusing" the D-Bus interface? I don't know of any app (beyond rockpool itself) using the interface yet, and normally, apps should push pins through Rebble's server.

**Comment by DylanVanAssche** [comment]
Sure, it would be awesome to see support for Pebble watches and to implement notifications for delays and cancellations. I'm open for suggestions, PRs, ...

**Comment by rubdos** [comment]
> Sure, it would be awesome to see support for Pebble watches and to implement notifications for delays and cancellations. I'm open for suggestions, PRs, ...

Awesome! If @abranson is okay with abusing his D-Bus, you can expect a PR.

Do you by any chance own a Pebble, Dylan? If you'd be interested, I might repair one more and hand it to you next FOSDEM :-)

**Comment by DylanVanAssche** [comment]
I don't own one, I'm waiting for the PineTime at the moment. I'm not sure about the support for SFOS at the moment though of the PineTime.

I will be at the next FOSDEM :) You have broken Peddles?

**Comment by rubdos** [comment]
> I will be at the next FOSDEM :) You have broken Peddles?

I broke one's screen while taking out a blown battery... Actually thinking to make a repair-business out of it. If I can get my hands on a working screen, I'll repair the one I have now.

**Comment by DylanVanAssche** [comment]
There are probably some vendors that sell parts for these :)

**Comment by abranson** [comment]
By all means hack away, would love to see native SFOS apps messing around with timeline pins - I'm not a fan on relying on cloud servers for distribution of your own data. However, this sounds more like you should be inserting events into the calendar, which should then be pushed to the Pebble timeline by Rockpool automatically. The benefit then is that they can be synced to remote calendars. I know Fahrplan has this functionality - you do need a privileges files to be able to interact with the SFOS calendar, but that's quite straightforward. There's one in Rockpool already.

**Comment by rubdos** [comment]
I'll have a look at how Fahrplan does it when I start this, because I don't immediately see the advantage, except for travels in the future.

Thanks for the blessing!

**Comment by rubdos** [comment]
> I'll have a look at how Fahrplan does it when I start this, because I don't immediately see the advantage, except for travels in the future.

![Fahrplan calendar entry](https://user-images.githubusercontent.com/1263440/69078700-b3d8d600-0a38-11ea-8af0-861b1323f77b.png)

I don't like it, I'd like to have my different trains as "I need to get of and change to platform x", and the calendar entry is not dynamic (trains in Belgium have a lot of change in their schedule). It's a cool feature for when you plan a larger travel ahead though.

If I get to implement and test it (when I finally have my Pebble), I'll post a video of how it looks. Maybe it's interesting to get in Fahrplan too!

**Comment by DylanVanAssche** [comment]
I agree, the Fahrplan approach doesn't suit the need of Belgian train commuters :)
